src/main/java/org/disq_bio/disq/HtsjdkReadsRddStorage.java

```diff
@@ -0,0 +1,280 @@
+package org.disq_bio.disq;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.util.Locatable;
+import java.io.IOException;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.disq_bio.disq.impl.file.FileSystemWrapper;
+import org.disq_bio.disq.impl.file.HadoopFileSystemWrapper;
+import org.disq_bio.disq.impl.file.NioFileSystemWrapper;
+import org.disq_bio.disq.impl.formats.bam.BamSink;
+import org.disq_bio.disq.impl.formats.bam.BamSource;
+import org.disq_bio.disq.impl.formats.cram.CramSink;
+import org.disq_bio.disq.impl.formats.cram.CramSource;
+import org.disq_bio.disq.impl.formats.sam.AbstractSamSink;
+import org.disq_bio.disq.impl.formats.sam.AbstractSamSource;
+import org.disq_bio.disq.impl.formats.sam.AnySamSinkMultiple;
+import org.disq_bio.disq.impl.formats.sam.SamFormat;
+import org.disq_bio.disq.impl.formats.sam.SamSink;
+import org.disq_bio.disq.impl.formats.sam.SamSource;
+
+/** The entry point for reading or writing a {@link HtsjdkReadsRdd}. */
+public class HtsjdkReadsRddStorage {
+
+  /** An option for configuring how to write a {@link HtsjdkReadsRdd}. */
+  public interface WriteOption {}
```

Comment:
I would suggest, to keep the classes more readable, to pull out the `ReadsWriteOption` interface and add the enums there instead of form part of the storage.

Reply:
I'm not sure about this - the options are tightly tied to the storage class.

Reply:
Several reasons why I suggested this:
* Using the methods here would require to add the full signature (e.g., `HtsjdkReadsRddStorage.FormatWriteOption.BAM`), which is too verbose from my point of view.
* Dowstream projects could define their own storage and want to use the same write-options for compatibility purposes.
* If different read storages are implemented (as you told me that it's the reason for the prefix), it will be nice to share writing options between them. Like that, they could be interchangeable.

Reply:
And I forgot to mention:
* TempPartsDirectory and FileCardinality options are tight to the storage object, but might be shared between different storages.